smarts/env/gymnasium/wrappers/metric/completion.py

```diff
@@ -113,7 +113,16 @@ def _get_dist(start: Point, end: Point) -> float:
                 point_b,
                 point_a,
             )
-
+        else:
+            raise
+    except CompletionError:
+        dist_tot = 1e10
+        logger.info(
+            "completion.get dist(): Did not find a route from "
+            "%s to %s, because too far off road. Score set to minimum.",
+            point_a,
+            point_b,
+        )
```

Comment:
Previously, after considering the following,
+ Vehicle might not be more than 5 meters off-road from nearest road as 150km/h is 4.2m in 0.1s.
+ All dead-end roads should be two-way roads, since the cars entering them have to exit somehow.

I thought the first distance computation (i.e., dist_tot = _get_dist(point_a, point_b)) is sufficient by itself (perhaps with a slightly increased radius=7), even without the need for the secondary reversed route computation. In short, realistically we should always be able to compute the remainder-distance cost without errors.

I think the error due to cars appearing spuriosly far off-road should either be avoided or handled elsewhere, as handling the repercussions here affects the realism of our cost computation code. Moreover, allowing this far off-road vehicle to propagate inside our cost computation code makes the values of other costs computed questionable.

Reply:
I would say primarily that it is unacceptable for the metrics to cause the program to crash.

It is also not that vehicles appear far off-road. Vehicles can drive there naturally if off-road is not set to cause done, especially when I revert the distance check for finding the route back to 5 meters. 

There may not be a realistic way of calculating the remainder-distance cost when off-road without introducing a logical error because the metric has already assumed that the agent is following the road graph. We can always increase the radius in which the vehicle can find a route from off-road but that adds in performance inefficiency and still does not solve the issue that the agent has escaped the graph (and potentially attempted to cheat the graph.)

To handle a failure of the metrics from a higher level, the caller would need to know about the radius restrictions that the metric has (e.g. 5 meters).

For now I have added a warning specifically mentioning this.

Reply:
+ `Metrics` verifies the environment suitability during instantiation. 
+ Here, `Metrics` cannot be used on an environment which has an agent with `DoneCriteria(off_road=False)`. 
+ On another comment, since environment suitability is checked by `Metrics`, I thought we don't have to call it `CompetitionMetrics`.

https://github.com/huawei-noah/SMARTS/blob/5d2dff7e0a91454cfd4722481ebfe5d99a298b29/smarts/env/gymnasium/wrappers/metrics.py#L276-L307

Reply:
Odd, somehow an agent was able to drive off-road without terminating. I think this is a bug.